Production_scripts/Social_Care/Care_Home.R

```diff
@@ -0,0 +1,299 @@
+  left_join(period_dates, by = c("period")) %>%
+  # filter out any episodes where discharge is before admission
+  mutate(dis_before_adm = ch_admission_date > ch_discharge_date & !is.na(ch_discharge_date)) %>%
+  filter(dis_before_adm == FALSE) %>%
+  # sort data
+  arrange(sending_location, social_care_id, period, ch_admission_date, ch_discharge_date)
+
+
+# Match with demographics data  ---------------------------------------
+
+# read in demographic data
+sc_demog <- haven::read_sav(get_sc_demog_lookup_path()) # remaining here until .rds file ready
+sc_demog <- readr::read_rds(get_sc_demog_lookup_path())
+
+matched_ch_data <- ch_clean %>%
+  left_join(sc_demog, by = c("sending_location", "social_care_id"))
+
+
+# Data Cleaning Care Home Name and Postcode ---------------------------------------
+
+matched_ch_clean <- matched_ch_data %>%
+  # correct postcode formatting
+  mutate(across(contains("postcode"), .x = format_postcode(.x)))
+
+
+# postcode lookup
+valid_spd_postcodes <- haven::read_sav(get_slf_postcode_path()) %>%
+  pull(postcode)
+
+
+# CH lookup
+ch_lookup <- readxl::read_xlsx(get_slf_ch_name_lookup_path()) %>%
+  rename(
+    ch_postcode = "AccomPostCodeNo",
+    ch_name_lookup = "ServiceName"
+  ) %>%
+  select(
+    ch_postcode,
+    ch_name_lookup,
+    DateReg,
+    DateCanx
+  ) %>%
+  # format postcode and CH name
+  mutate(ch_postcode = format_postcode(ch_postcode),
+         ch_name_lookup = toupper(ch_name_lookup))
+
+
+name_postcode_clean <- matched_ch_clean %>%
+  # deal with capitalisation of CH names
+  mutate(ch_name = stringr::str_to_title(ch_name)) %>%
+  # deal with punctuation in the CH names
+  mutate(ch_name = stringr::str_replace_all(ch_name, "[[:punct:]]", " ")) %>%
+  # replace invalid postcode with NA
+  mutate(ch_postcode = na_if(ch_postcode, ch_postcode %in% valid_spd_postcodes)) %>%
+  # where there is only a single Care Home at the Postcode, use that name
+  group_by(ch_postcode) %>%
+  # fill in names where NA but has a name in another record
+  fill(ch_name, .direction = "downup") %>%
+  # change name if CH name does not match previous name
+  mutate(ch_name_change_counter = pmax(ch_name != lag(ch_name) & !is.na(ch_name), FALSE, na.rm = TRUE)) %>%
+  mutate(ch_name = if_else(ch_name_change_counter == 1, last(ch_name), ch_name)) %>%
+  ungroup() %>%
+  # match CH names with CH lookup
+  left_join(ch_lookup, by = c("ch_postcode")) %>%
+  # check CH name matches the lookup name
+  mutate(name_match = if_else(ch_name != ch_name_lookup | is.na(ch_name_lookup), 0, 1)) %>%
+  # replace ch name with lookup name if not a match
+  mutate(ch_name = if_else(name_match == 0 & !is.na(ch_name_lookup), ch_name_lookup, ch_name)) %>%
+  # check admission date within CH dates
+  mutate(date_check = if_else(is.na(DateReg) | is.na(DateCanx) | ch_admission_date %in% range(DateReg, DateCanx), TRUE, FALSE))
+
+
+# Data Cleaning Care Home Data ---------------------------------------
+
+ch_data_clean <- matched_ch_data %>%
```

Comment:
Should this next section use the 
```suggestion
name_postcode_clean <- matched_ch_clean %>%
  # deal with capitalisation of CH names
  mutate(ch_name = stringr::str_to_title(ch_name)) %>%
  # deal with punctuation in the CH names
  mutate(ch_name = stringr::str_replace_all(ch_name, "[[:punct:]]", " ")) %>%
  # replace invalid postcode with NA
  mutate(ch_postcode = na_if(ch_postcode, ch_postcode %in% valid_spd_postcodes)) %>%
  # where there is only a single Care Home at the Postcode, use that name
  group_by(ch_postcode) %>%
  # fill in names where NA but has a name in another record
  fill(ch_name, .direction = "downup") %>%
  # change name if CH name does not match previous name
  mutate(ch_name_change_counter = pmax(ch_name != lag(ch_name) & !is.na(ch_name), FALSE, na.rm = TRUE)) %>%
  mutate(ch_name = if_else(ch_name_change_counter == 1, last(ch_name), ch_name)) %>%
  ungroup() %>%
  # match CH names with CH lookup
  left_join(ch_lookup, by = c("ch_postcode")) %>%
  # check CH name matches the lookup name
  mutate(name_match = if_else(ch_name != ch_name_lookup | is.na(ch_name_lookup), 0, 1)) %>%
  # replace ch name with lookup name if not a match
  mutate(ch_name = if_else(name_match == 0 & !is.na(ch_name_lookup), ch_name_lookup, ch_name)) %>%
  # check admission date within CH dates
  mutate(date_check = if_else(is.na(DateReg) | is.na(DateCanx) | ch_admission_date %in% range(DateReg, DateCanx), TRUE, FALSE))


# Data Cleaning Care Home Data ---------------------------------------

ch_data_clean <- name_postcode_clean %>%
```
Should this next section take the object `name_postcode_clean`? i'm not seeing where the ch name changes are used in the script further down

Reply:
changed for now - worth checking with @Moohan ?